### ISSUE TITLE
error on compile

### ISSUE THREAD

**Issue by yo2ldk** [ZalophusDokdo/ESP8266_LED_Matrix_Bar_Weather_Station#1]
Hello,
Thank you for sharing this nice project, I search from some time a clock with Meteo posibilities and yours look very nice; what Arduino version software you use and Arduinojson version ?
`unfortunately I cannot compile it at all If you can help me with any tips or advice will be great:
           `
                 from C:\Users\yo2ldk\Desktop\ESP8266_LED_Matrix_Bar_Weather_Station\ESP8266_LED_Matrix_Bar_Weather_Station.ino:42:

C:\Users\yo2ldk\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.5.0\cores\esp8266/core_esp8266_si2c.c: In function 'void twi_init(unsigned char, unsigned char)':

C:\Users\yo2ldk\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.5.0\cores\esp8266/core_esp8266_si2c.c:149:40: error: invalid conversion from 'void (*)()' to 'void (*)(void*)' [-fpermissive]

   ets_timer_setfn(&timer, onTimer, NULL);

                                        ^

In file included from C:\Users\yo2ldk\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.5.0\cores\esp8266/pgmspace.h:11:0,

                 from C:\Users\yo2ldk\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.5.0\cores\esp8266/Arduino.h:258,

                 from sketch\ESP8266_LED_Matrix_Bar_Weather_Station.ino.cpp:1:

C:\Users\yo2ldk\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.5.0/tools/sdk/include/ets_sys.h:195:6: error:   initializing argument 2 of 'void ets_timer_setfn(ETSTimer*, void (*)(void*), void*)' [-fpermissive]

 void ets_timer_setfn(ETSTimer *t, ETSTimerFunc *fn, void *parg);

      ^

In file included from sketch\DHT22.h:14:0,

                 from C:\Users\yo2ldk\Desktop\ESP8266_LED_Matrix_Bar_Weather_Station\ESP8266_LED_Matrix_Bar_Weather_Station.ino:49:

sketch\DHT12.h: At global scope:

C:\Program Files (x86)\Arduino\libraries\DHT-sensor-library-master/DHT.h:33:15: error: expected identifier before numeric constant

 #define DHT12 12

               ^

sketch\DHT12.h:55:7: note: in expansion of macro 'DHT12'

 class DHT12 {

       ^

C:\Program Files (x86)\Arduino\libraries\DHT-sensor-library-master/DHT.h:33:15: error: expected unqualified-id before numeric constant

 #define DHT12 12

               ^

sketch\DHT12.h:55:7: note: in expansion of macro 'DHT12'

 class DHT12 {`

**Comment by sandro-defender** [comment]
same problem :(

**Comment by pangit99** [comment]
Similar one here. I've tried refreshing all the libraries twice, no go. Anything else it could be?

**Comment by hawk88** [comment]
Yes. The same error in compiling problem for everyone. It is because the use of #ifdef that causes the problem. Although the coding becoming selectable for multi sensors, multi environment, but the drawback is the increasing complexity of the coding.

Can you update the codes with the latest version?

Thank you very much for sharing your codes.

**Comment by djmitho** [comment]
Same problem compiling with core_esp8266_si2c.c
Any solution?
Thanks and good job!!

**Comment by hawk88** [comment]
Check all the header ".h" files...  In Config.h there are 2 lines (line #12 #13):
#ifndef CONFIG_H
#define CONFIG_H
They are intentionally left reversed.
They should be as follow:
#define CONFIG_H
#ifndef CONFIG_H
If you have #define CONFIG_H, then #ifndef CONFIG_H follows. The compiler will then use whatever follow after #ifndef CONFIG_H to compile.
All other ''.h" files having **#ifndef** and **#define** in reversed order, and you can change them according to your need.

**Comment by djmitho** [comment]
Thank you for your answer!!
After make changes i have now another error and i don't know why

WiFi_Setup.h: In function 'void wifi_start()':
WiFi_Setup.h:22:21: error: 'wifimanager_start' was not declared in this scope
   wifimanager_start();

do you you know what is it?

**Comment by hawk88** [comment]
You need to study overall coding carefully and understand what the coding are all about.
Examine all the functions defined in each related files.

For the following codes:

   void wifi_start() {
     wifimanager_start();

Study all about wifimanager carefully how to use wifimanager library.
You can study from the reference by following this link:  https://github.com/tzapu/WiFiManager

**Comment by ZalophusDokdo** [comment]
Explained in README.md.
